### PR TITLE
Check status by FileUpload uuid (bug 1223137)

### PR DESCRIPTION
**Pull request by mstriemer**
https://bugzilla.mozilla.org/show_bug.cgi?id=1223137

The tests are a bit crazy but it's hard to create these objects so that's how it ended up =\

### CODE REVIEW

File: apps/signing/tests/test_views.py [hunk 1]
```diff
@@ -168,6 +172,50 @@ def test_version_exists(self):
         assert response.status_code == 200
         assert 'processed' in response.data
 
+    @mock.patch('devhub.tasks.Version.from_upload')
+    def test_version_exists_with_pk(self, version_from_upload):
+        # Mock submit_file so the version won't be created.
```
Comment (mstriemer) [inline apps/signing/tests/test_views.py]: This comment is out of date. I'll update it.

File: docs/topics/api/signing.rst [hunk 2]
```diff
@@ -98,9 +99,10 @@ property.
                     }
                 ],
                 "passed_review": true,
+                "pk": "f68abbb3b1624c098fe979a409fe3ce9",
                 "processed": true,
                 "reviewed": true,
-                "url": "https://addons.mozilla.org/api/v3/addons/%40new-unlisted-api/versions/1.0/",
+                "url": "https://addons.mozilla.org/api/v3/addons/%40new-unlisted-api/versions/1.0/uploads/f68abbb3b1624c098fe979a409fe3ce9/",
```
Comment (kumar303) [inline docs/topics/api/signing.rst]: just a minor detail (and I know it's from another commit) but `@new-unlisted-api` is confusing as an example because it looks like a semantic URL qualifier saying "this add-on is unlisted." I'd suggest `@example-guid` or something more neutral.

File: apps/signing/tests/test_views.py [hunk 3]
```diff
@@ -168,6 +172,50 @@ def test_version_exists(self):
         assert response.status_code == 200
         assert 'processed' in response.data
 
+    @mock.patch('devhub.tasks.Version.from_upload')
+    def test_version_exists_with_pk(self, version_from_upload):
+        # Mock Version.from_upload so the version won't be created.
+        self.create_version('3.0')
```
Comment (kumar303) [inline apps/signing/tests/test_views.py]: this might be a little cleaner:
````
with mock.patch('devhub.tasks.Version.from_upload') as version_from_upload:
    self.create_version(...)
````

Comment (kumar303) [inline apps/signing/tests/test_views.py]: The fact that you need a mock here suggests to me that you should be breaking `create_vesion()` down into more of a factory approach:
````
addon = Addon.objects.create(...)
version = ...
file = ...
````
I know that's a lot more work and doesn't need to come in this patch. I think it highlights why relying on HTTP requests for test setup can be awkward (it's also usually a lot slower).

File: apps/signing/views.py [hunk 4]
```diff
@@ -81,14 +81,14 @@ def put(self, request, addon, version_string):
...
+        file_upload_qs = FileUpload.objects.filter(
+            addon=addon, version=version_string)
         try:
-            file_upload = (FileUpload
-                           .objects
-                           .order_by('-created')
-                           .filter(addon=addon, version=version_string)
-                           [:1]
-                           .get())
+            if pk is None:
```
Comment (kumar303) [inline apps/signing/views.py]: logging for each of these if branches would be nice. Like: `getting latest upload for {addon} {version}: {upload.pk}`

**Comment by kumar303** [comment]
r+wc

File: apps/signing/tests/test_views.py (hunk 3 above)
Comment (mstriemer) [inline apps/signing/tests/test_views.py]: Yeah using the view is definitely not ideal but there are a ton of edge cases around creating add-ons and versions and I wanted it to be as close to real as possible. The tests are definitely slow, unfortunately.
